src/main/java/org/jenkinsci/plugins/workflow/cps/FlowHead.java

```diff
@@ -114,29 +116,32 @@ void newStartNode(FlowStartNode n) throws IOException {
         execution.storage.storeNode(head, false);
     }
 
-    void setNewHead(FlowNode v) {
+    void setNewHead(@Nonnull FlowNode v) {
+        if (v == null) {
```

Comment:
So my brain briefly melts at the combination of `@Nonnull` and `if (v == null)` - I'm assuming this is defense-in-depth, with the `@Nonnull` covering compile-time scenarios and the `if (v == null)` covering even edgier-case scenarios?

Reply:
Yeah, it's protecting against some things that *shouldn't* happen but yet were still happening in the wild, even though Findbugs didn't find issues at compile-time (probably it wasn't digging deeply enough).

Reply:
Yeah, I have no problem with that approach - probably worth adding a comment explaining that, though. It'd be awfully easy for a later maintainer (i.e., me, because I'm forgetful) to not know/remember that and just think "Oh, that's a pointless null check, lemme clear the dead code".